### PR TITLE
Support FASTLY_SERVICE_ID environment variable

### PR DESCRIPTION
**Pull request by peterbourgon**
Addresses #301.

```
$ fastly logs tail --service-id abc
ERROR: the Fastly API returned 404 Not Found: Record not found (Cannot find service 'abc').

$ env FASTLY_SERVICE_ID=abc fastly logs tail
ERROR: the Fastly API returned 404 Not Found: Record not found (Cannot find service 'abc').

$ env FASTLY_SERVICE_ID=abc fastly logs tail --service-id def
ERROR: the Fastly API returned 404 Not Found: Record not found (Cannot find service 'def').
```

How would you like this to be tested?

### CODE REVIEW

File: pkg/backend/create.go [hunk 1]
```diff
@@ -27,7 +28,7 @@ func NewCreateCommand(parent cmd.Registerer, globals *config.Data) *CreateComman
 	c.Globals = globals
 	c.CmdClause = parent.Command("create", "Create a backend on a Fastly service version").Alias("add")
 
-	c.CmdClause.Flag("service-id", "Service ID").Short('s').Required().StringVar(&c.Input.ServiceID)
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').Envar(env.ServiceID).Required().StringVar(&c.Input.ServiceID)
```
Comment (peterbourgon) [inline pkg/backend/create.go:31]: Should we also call out the env var in the help text, as `-token` does?

**Comment by peterbourgon** [comment]
Or maybe helper function(s) would be better?

```go
func RegisterOptionalServiceID(c *kingpin.CmdClause, dst *string) {
    c.Flag("service-id", "Service ID").Short('s').Envar(env.ServiceID).StringVar(dst)
}

func RegisterRequiredServiceID(c *kingpin.CmdClause, dst *string) {
    c.Flag("service-id", "Service ID").Short('s').Envar(env.ServiceID).Required().StringVar(dst)
}
```

**Comment by Integralist** [comment]
@peterbourgon 

> Or maybe helper function(s) would be better?

I think I'd prefer that approach 👍🏻 

As a side note, in `main` there's been some changes where by the Service ID flag is no longer explicitly required. This was to support falling back to reading the value from the fastly manifest using the [`ServiceID()`](https://github.com/fastly/cli/blob/c8dbff77b7407272789ea3e1239320b0c254fcda/pkg/compute/manifest/manifest.go#L78-L88) function.

**Comment by Integralist** [comment]
I went ahead and implemented this feature here: https://github.com/fastly/cli/pull/320 as I need to cut a new CLI release soon and wanted this feature included in that release. This PR had looked to have gone stale with a bunch of changes in the `main` branch and so it was easier for me to implement this myself as I had the context around the changes that have been happening.

Thank you for the (internal) suggestion to implement this feature, I appreciate you taking the time to look into this.

**Comment by peterbourgon** [comment]
Oh sorry!! This had entered my Backburner and I let it slide :( Thank you!

**Pull request closed without merging**